ec2/server.py

```diff
@@ -1,29 +1,49 @@
 from flask import Flask
 import requests
+import os
 import boto3
-import pickle
+from dotenv import load_dotenv
 
-OLLAMA_URL = 'http://localhost:11434/api/generate'
+load_dotenv()
+
+OLLAMA_URL = "http://localhost:11434/api/generate"
 
 app = Flask(__name__)
-s3 = boto3.client('s3')
+s3 = boto3.client(
+    "s3",
+    aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
+    aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+    aws_session_token=os.getenv("AWS_SESSION_TOKEN"),
+)
+
 
 def load_data(bucket_name, file_name):
     obj = s3.get_object(Bucket=bucket_name, Key=file_name)
-    data = obj['Body'].read()
-    return pickle.loads(data, encoding='bytes')
+    data = obj["Body"].read()
+    return data.decode("utf-8")
```

Comment:
have we tested if this successfully reads input from an S3 bucket?

Reply:
Yessir it works!

Reply:
lesgooo